capstone/src/main/java/com/google/sps/Objects/TFIDFStringHelper.java

```diff
@@ -0,0 +1,100 @@
+package com.google.sps.Objects;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class containing String manipulation helper methods, used when calculating TF-IDF.
+ */
+public class TFIDFStringHelper {
+  
+  /**
+   * "Sanitizes" a String so it can be analyzed.
+   * Remove all punctuation and convert String to lowercase.
+   */
+  public static String sanitize(String input) {
+    if (input ==  null || input.length() == 0) {
+      return input;
+    }
+
+    input = input.toLowerCase();
+    // TO-DO: Handle splitting up sentences/phrases into multiple Strings
+    input = removeUrls(input);
+    // Regex identifies punctuation except apostrophes.
+    input = input.replaceAll("[\\p{P}&&[^']]", " ");
+    input = removeExtraWhitespaces(input);
+  
+    return input;  
+  }
+
+  /**
+   * Helper method for {@code sanitize()}. Removes urls from a String.
+   */
+  public static String removeUrls(String input) {
+    input = input.replaceAll("[\\S]+://[\\S]+", "");
+    input = input.replaceAll("www.[\\S]+", "");
+    return input;
+  }
```

Comment:
I think Andrew's comment wasn't specifically saying we need to sanitize URLs but rather giving an example of other punctuation that you would want to remove. From my understanding, this is the list of punctuation that will be removed: 
!"#$%&'()*+,-./:;<=>?@[]^_`{|}~
therefore I think what you have works and this isn't necessary. 

Also, I found the accepted answer and it's follow-up comments insightful: herehttps://stackoverflow.com/questions/18830813/how-can-i-remove-punctuation-from-input-text-in-java

Reply:
Though if you want to specifically remove URLs that is also fine :-)

Reply:
Yes I saw that thread! I ended up going with a similar solution as found in the comments.

Also I do think removing URLs makes sense, so I think I'll leave it.